easybuild/easyblocks/m/molpro.py

```diff
@@ -32,15 +32,31 @@
 import re
 import sys
 
+from easybuild.easyblocks.generic.binary import Binary
 from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.framework.easyconfig import CUSTOM
 from easybuild.tools.build_log import EasyBuildError
 from easybuild.tools.filetools import mkdir, read_file
-from easybuild.tools.run import run_cmd
+from easybuild.tools.run import run_cmd, run_cmd_qa
 
 
-class EB_Molpro(ConfigureMake):
+class EB_Molpro(Binary, ConfigureMake):
     """Support for building and installing Molpro."""
 
+    @staticmethod
+    def extra_options():
+        """Define custom easyconfig parameters for Molpro."""
+        # Combine extra variables from Binary and ConfigureMake easyblocks as
+        # well as those needed for Molpro specifically
+        extra_vars = Binary.extra_options()
+        extra_vars = ConfigureMake.extra_options(extra_vars)
+        extra_vars.update({
+            'license_token_present': [True, "Can we supply a license token at install time?", CUSTOM],
```

Comment:
is this custom parameter necessary? can't this be detected based on whether `self.license_token` is available (which is  required currently)

if you need support to install without a license token being available, just relax the requirements w.r.t. `self.license_token` instead (e.g. just issue a warning if it's not available), I would stay away from adding a custom parameter like this